### ISSUE TITLE
p_130086_ is null

### ISSUE THREAD

**Issue by ChecksumDev** [ricksouth/serilum-mc-mods#2095]
## **Information**
Minecraft version: 1.20.1
Modloader: NeoForge
NeoForge version: 47.1.81 (also tested on recommended Forge 1.20.1)
Environment: Multiplayer

Mod name: VillagerNames
Mod version: 7.1

## **Description**
When installed on the server (not singleplayer server), villagernames will prevent you from joining with the error message and (possibly) crash your game shortly after if you wait on the menu.

> net.minecraft.resources.resourcelocation.tostring() p_130086_ is null

For reference, this is the mod-list, though we thoroughly tested one by one finding this mod to be the culprit for around 6 hours.
https://github.com/ChecksumDev/velvet_modpack/tree/69ab565f66920517169bf6cdf7ea946bc5d6a510

**Comment by ricksouth** [comment]
Apologies for the inconvenience, that's not a good situation. Could you upload the crash log?

**Comment by ricksouth** [comment]
As I cannot find any reference to a toString() on a ResourceLocation within Villager Names, I'll need the logs. I'll take a look as soon as I can when received. Thanks!

**Comment by ChecksumDev** [comment]
I'm not entirely sure what I'd be looking for, the server only disconnects the client with that message as the disconnect message. It is seemingly unrelated to a villager being nearby (within simulation distance). I will see if I can get the crash log of the client later today when it closes abruptly after 30 secs or so, but I am unsure if it will be related or just another mod getting angry because of something happening here.

**Comment by ChecksumDev** [comment]
Also, the mod was previously installed prior to the 7.1 version with no issues; It's likely the problem was introduced in this version through some means.

**Comment by robosoup2** [comment]
I am also having this issue, removing the mod didn't fix it though :( did you also have to delete and remake the world or no?

**Comment by ChecksumDev** [comment]
> I am also having this issue, removing the mod didn't fix it though :( did you also have to delete and remake the world or no?

For me deleting the mod did fix it and there was no world remaking, there might be a mod going around causing an incompatibility with mods including this one, though it wouldn't make sense.

**Comment by ChecksumDev** [comment]
@ricksouth I have attached the client crash I get shortly after I get the disconnect message.
[crash-2023-12-10_18.33.12-client.txt](https://github.com/ricksouth/serilum-mc-mods/files/13640874/crash-2023-12-10_18.33.12-client.txt)

**Comment by ChecksumDev** [comment]
@ricksouth @robosoup2 If you're looking for an incompatibility, this was the pack update that broke villager name's compatibility, if not this mod its __likely__ one of the others in this list (on the left) causing the incompatibility.  

https://github.com/ChecksumDev/velvet/commit/1a6269d48118621cbdbd6565045290c6d920e6ee

**Comment by robosoup2** [comment]
btw turns out my issue was a new version of Waystones with the same bug
(that might not be relevant or helpful sorry, but i'm not a mod dev so i wouldn't know)

**Comment by ricksouth** [comment]
This has been (hopefully) fixed in version 8.22 of Collective. Villager Names itself doesn't need an update. Please let me know if you're still having problems, preferably by opening a new issue so I don't miss it. Sorry it sat for a while, it's been difficult keeping up.

Thank you for opening the issue and giving the mod a try!

https://curseforge.com/minecraft/mc-mods/collective, https://modrinth.com/mod/collective
